### DISCUSSION TITLE
请问Magpie是用什么UI框架开发的？想学习一下。

### DISCUSSION THREAD

**Discussion by Shiiho11** [Blinue/Magpie#898] [Q&A] [answered]
大佬的Magpie写的很好，想学习一下，希望能看懂代码，自己写点小工具或者有机会的话贡献一点。
之前没接触过客户端开发，完全看不懂项目结构。
看着好像是用了winui，就去看了一下winui3（基本上只有C#教程），好像和Magpie使用方式也有些不太一样。
了解之后重新回头看了一下，发现Magpie似乎用的winui2（uwp）？

请问一下Magpie是用什么UI框架开发的？怎么入门（希望能从看懂目录结构或者Main开始）？

谢谢。

**Comment by Blinue** [comment, marked as answer]
UI 框架是 XAML Islands + WinUI 2 + C++/WinRT。这套框架对于初学者是非常困难的，必须有 C++ 和 Win32 开发经验才行。我把 Magpie 的 UI 框架提取成了单独的项目：[Xaml-Islands-Cpp](https://github.com/Blinue/Xaml-Islands-Cpp)，你可以从这里入手。

**Reply by Shiiho11** [reply]
谢谢大佬。
